### PR TITLE
fix(operator): Add compatibility with the last operator changes

### PR DESCRIPTION
**Pull request by davidfestal**
This PR adds compatibility  with the last Che operator changes done in PR https://github.com/eclipse/che-operator/pull/51

This adds support for deploying registries along with the Che server from the operator.

### CODE REVIEW

**Comment by benoitf** [comment]
FYI semantic pr check is failing

**Comment by benoitf** [comment]
https://github.com/che-incubator/chectl/blob/master/README.md#providing-pull-request

**Comment by benoitf** [comment]
FYI would be nice to have `fix(operator): Add compatibility with the last operator changes` in commit message (to have something in parenthesis that describe the area of the fix/enhancement (it's just a FYI)

**Comment by benoitf** [comment]
associated pull request https://github.com/eclipse/che-operator/pull/51 has no linked issue
so I don't know the associate issue, to know if it's targeting `7.0.0` or not ?

File: src/api/kube.ts [hunk 1]
```diff
@@ -745,20 +745,31 @@ export class KubeHelper {
...
+      yamlCr.spec.k8s.ingressDomain = flags.domain
+      let pluginRegistryUrl = flags['plugin-registry-url']
+      if (pluginRegistryUrl) {
+        yamlCr.spec.server.pluginRegistryUrl = pluginRegistryUrl
+      }
+      let devfileRegistryUrl = flags['devfile-registry-url']
+      if (devfileRegistryUrl) {
+        yamlCr.spec.server.devfileRegistryUrl = devfileRegistryUrl
+      }
+      yamlCr.spec.auth.identityProviderImage = 'eclipse/che-keycloak:' + yamlCr.spec.server.cheImageTag
```
Comment (benoitf) [inline src/api/kube.ts]: please use constant or string litterals 
```
`eclipse/che-keycloak:${yamlCr.spec.server.cheImageTag}`

**Comment by davidfestal** [comment]
> FYI would be nice to have fix(operator): Add compatibility with the last operator changes in commit message (to have something in parenthesis that describe the area of the fix/enhancement (it's just a FYI)

@benoitf done

**Comment by davidfestal** [comment]
> associated pull request eclipse/che-operator#51 has no linked issue and no description at all
so I don't know the associate issue, to know if it's targeting 7.0.0 or not ?

I updated the description of PR eclipse/che-operator#51 to link to the issues that are part of the endgame plan.
Sorry about that.

**Comment by benoitf** [comment]
@davidfestal thak you so much. You should update the commit messages as well to be really compliant with conventional commits (it seems the check is ok even if messages are not following it)

@davidfestal and thanks for the endgame issue

**Comment by davidfestal** [comment]
@benoitf @l0rd I force pushed the rebase of this PR upon the last master, and also got rid of string literals (which in fact were not useful)
Could review a last time ?

File: src/api/kube.ts [hunk 2]
```diff
@@ -745,20 +745,36 @@ export class KubeHelper {
...
+      let pluginRegistryUrl = flags['plugin-registry-url']
+      if (pluginRegistryUrl) {
+        yamlCr.spec.server.pluginRegistryUrl = pluginRegistryUrl
+      }
+      let devfileRegistryUrl = flags['devfile-registry-url']
+      if (devfileRegistryUrl) {
+        yamlCr.spec.server.devfileRegistryUrl = devfileRegistryUrl
+      }
+      const tagExp = /:[^:]*$/
+      const newTag = ':' + yamlCr.spec.server.cheImageTag
```
Comment (benoitf) [inline src/api/kube.ts]: why concatenate ? ;-)

Comment (davidfestal) [reply src/api/kube.ts]: Oh, just because I'm not used to the typescript syntax, and did't pay attention. Let me fix it !

Comment (davidfestal) [reply src/api/kube.ts]: Fxed in commit https://github.com/che-incubator/chectl/pull/232/commits/8e1f94a4e315bfd5f729f70e46abf7df5b125ef4

**Comment by davidfestal** [comment]
@l0rd @benoitf Do you think you could merge this PR if it's OK for you ? It seems I don't have rights to do it myself.

**Comment by benoitf** [comment]
hmm will fixe that

**Comment by benoitf** [comment]
@davidfestal please check

**Pull request merged**